api/src/main/java/bio/terra/tanagra/indexing/job/BuildTextSearchStrings.java

```diff
@@ -0,0 +1,84 @@
+package bio.terra.tanagra.indexing.job;
+
+import bio.terra.tanagra.exception.InvalidConfigException;
+import bio.terra.tanagra.indexing.EntityJob;
+import bio.terra.tanagra.query.Query;
+import bio.terra.tanagra.underlay.DataPointer;
+import bio.terra.tanagra.underlay.Entity;
+import bio.terra.tanagra.underlay.TablePointer;
+import bio.terra.tanagra.underlay.datapointer.BigQueryDataset;
+import bio.terra.tanagra.utils.GoogleBigQuery;
+import com.google.cloud.bigquery.Table;
+import com.google.cloud.bigquery.TableId;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BuildTextSearchStrings extends EntityJob {
+  private static final Logger LOGGER = LoggerFactory.getLogger(WriteParentChildIdPairs.class);
+
+  public BuildTextSearchStrings(Entity entity) {
+    super(entity);
+  }
+
+  @Override
+  public String getName() {
+    return "BUILD TEXT SEARCH (" + getEntity().getName() + ")";
+  }
+
+  @Override
+  public void dryRun() {
+    run(true);
+  }
+
+  @Override
+  public void run() {
+    run(false);
+  }
+
+  private void run(boolean isDryRun) {
+    Query selectIdTextPairs = getEntity().getSourceDataMapping().queryTextSearchString();
+    String sql = selectIdTextPairs.renderSQL();
+    LOGGER.info("select id-text pairs SQL: {}", sql);
+
+    BigQueryDataset outputBQDataset = getOutputDataPointer();
```

Comment:
Can some of the code in these functions be refactored? It seems like they were very similar (identical?) across the last three PRs.

Reply:
Good idea. Moved chunks of this code into the base class.